src/sh.janet

```diff
@@ -1102,6 +1102,29 @@
     :error nil
   })
 
+(defn- make-history-builtin
+  []
+  @{
+    :pre-fork
+      (fn builtin-history [self args] nil)
+    :post-fork
+      (fn builtin-history
+        [self args]
+        (let [hist-file (-> (fiber/current)
```

Comment:
@andrewchambers please check this. feels hacky, but couldn't find a different way.

Reply:
I think the correct way is to use the history api from readline - https://tiswww.case.edu/php/chet/readline/history.html#SEC6

So it requires some C programming, or we rewrite our own version of readline in janet.

Reply:
Will it work with libnoise or libedit?

Reply:
libedit follows the same api yes. libnoise it needs to be written, or just do nothing, (janet error may be ok) but still compile.